### ISSUE TITLE
dnx run command fails when user home directory contains space.

### ISSUE THREAD

**Issue by jrieken** [OmniSharp/omnisharp-vscode#28]
_From @retk on February 3, 2016 14:40_

Hi!
Unfortunately my home directory contains a space character: 'C:\Users\Ersek Attila' 

When I try to run any asp.net core application with the dnx run web command, i get the following error in the command window:
'C:\Users\Ersek' is not recognized as an internal or external command,...

vscode version: 0.10.6

_Copied from original issue: Microsoft/vscode#2674_

**Comment by retk** [comment]
There's already a pull request for the run-in-terminal package that fixes this issue on a lower level:
https://github.com/Microsoft/run-in-terminal/pull/2

**Comment by jtsom** [comment]
Any idea when a fix will be made so it get through to VS Code? Using dnx commands are impossible through VS Code until this is corrected, making VSCode useless for targeting .NET Core on Windows.

**Comment by egamma** [comment]
@retk the PR has been merged.

Ping @DustinCampbell @davidfowl

**Comment by jtsom** [comment]
And what is necessary to get this to actually be in VS Code?

**Comment by jrieken** [comment]
the `run-in-terminal@0.0.3` contains the fix and omnisharp ships with that version of the extension. There might be another issue somewhere along the way.

**Comment by DustinCampbell** [comment]
Omnisharp doesn't ship with `run-in-terminal` -- did you mean the C# Extension?

**Comment by jrieken** [comment]
Yeah, I meant the extension

**Comment by jtsom** [comment]
Either way... it's not in VS Code :)

**Comment by retk** [comment]
run-in-terminal is in package dependencies

```
  "dependencies": {
   ...
    "run-in-terminal": "*",
   ...
  },
```

although as @jrieken mentioned there might be another issues too

**Comment by jtsom** [comment]
Well, then I've opened a new issue on VS Code. If the dependencies have been fixed, then something in Code is broken.

**Comment by jtsom** [comment]
And it was closed and moved back here. 

Is anyone going to take this up and make sure it gets to VS Code??? I don't even know where the problem is anymore... VS Code? OmniSharp? run-in-terminal? Everyone is pointing to the other...

**Comment by DustinCampbell** [comment]
This is the right place for the bug.

**Comment by retk** [comment]
i'll try to reproduce the issue later today with vscode 1.0 and update the issue with details.

**Comment by DustinCampbell** [comment]
Note that "dnx run" is unlikley to get fixed. We're focused on supporting ASP .NET Core RC2. When that is released, DNX will go away.

**Comment by retk** [comment]
@DustinCampbell : yes, i'm aware of that, although the csharp package 0.3.7 still depends on run-in-terminal package. it might use this dependency for other reasons too.

I've managed to reproduce the issue with vscode 1.0. It installs csharp package 0.3.7 as extension into:
~/.vscode\extensions\ms-vscode.csharp-0.3.7

run-in-terminal package version 0.0.3 is installed as lates version of the dependency (see package.json under dependencies)

Although this version has the commited pull request the problem still exists. To fix the issue the following changes are needed:

diff --git a/lib/index.ts b/lib/index.ts
index 09c5128..e802873 100644
--- a/lib/index.ts
+++ b/lib/index.ts
@@ -60,7 +60,7 @@ function runInTerminalWin(file: string, args: string[], options: Options): Promi
         '/wait',
         'cmd.exe',
         '/c',
-        `"${file} ${args.map(escapeWinArg).join(' ') } & pause"`
+        `""${file}" ${args.map(escapeWinArg).join(' ') } & pause"`
     ];

and it must be recompiled with tsc.

Should I open a new issue in run-in-terminal and create a pull request? 

I'm not sure if this issue must remain open here until a new version is released, or someone should open a new one somewhere else? (for eg the cshap package is affected as well)

**Comment by DustinCampbell** [comment]
FWIW, I'm planning to fix this by not using run-in-terminal at all. I think it's a bit awkward that a new terminal window is created for this operation. Instead, I'll just launch "dotnet" as a child process and capture the output and display it within VS Code.

**Comment by DustinCampbell** [comment]
This should be fixed with #238. While the "dnx run" command is no longer relevant, we have removed run-in-terminal so other commands should not exhibit this problem.
